### ISSUE TITLE
Fix build on OS X El Capitan. [phab:D579]

### ISSUE THREAD

**Issue by kythe** [google/kythe#827]
_Created by **zarko** at 2015-10-26 23:56:29_: 

This CL addresses some changes made in El Capitan: most notably,
OpenSSL was removed and system headers were moved around.
Autodetecting the correct location to find system headers is left
to future work (it's unclear whether this is something that
needs fixed in upstream LLVM, since we're getting the wrong defaults
for Clang's header search configuration).

I've also added a switch to kindex_tool to rename transcript
hashes that should be stable across platforms and code changes.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-10-27 16:49:07_:

Bazel changes LGTM; leaving C++ to fromberger

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2015-10-27 16:49:07_

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-10-27 16:51:42_:

Also, can you look into why the linter is wrong here?  It works correctly on my end.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-10-27 16:57:42_:

Did you test on OS X? It looks like the lint check uses grep in a way that doesn't work there:

```
$ grep -Pq 'Copyright 201[45] Google Inc. All rights reserved.' kythe/cxx/indexer/cxx/IndexerASTHooks.cc
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
```

**Comment by kythe** [comment]
_**zarko** wrote at 2015-10-27 17:09:51_:

  - Get rid of an incompatible flag.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2015-10-28 18:04:48_:

"Identifiers" is sufficiently overloaded that I'd suggest a different word here—e.g., "hash codes" or "sequence numbers" or something more specific.

**Comment by kythe** [comment]
_**fromberger** wrote at 2015-10-28 18:04:48_:

The C++ stuff looks fine, apart from one documentation-related nit.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-10-28 19:06:45_:

  - Fix documentation.

**Comment by kythe** [comment]
_**zarko** wrote at 2015-10-28 19:11:01_:

  - clang-format nit

**Comment by kythe** [comment]
_**zarko** closed via commit 78d6de9019368f95e4f0be86835a908bd7ff9331 at 2015-10-28 19:11:24_
